.github/workflows/esrupgrade-common.yml

```diff
@@ -0,0 +1,152 @@
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Run docker compose
+        run: |
+          cd build
+          docker-compose --no-ansi run --rm start_dependencies
+          cat ../tests/test-data.ldif | docker-compose --no-ansi exec -T openldap bash -c 'ldapadd -x -D "cn=admin,dc=mm,dc=test,dc=com" -w mostest';
+          docker-compose --no-ansi exec -T minio sh -c 'mkdir -p /data/mattermost-test';
+          docker-compose --no-ansi ps
+      - name: Wait for docker compose
+        run: |
+          until docker network inspect ghactions_mm-test; do echo "Waiting for Docker Compose Network..."; sleep 1; done;
+          docker run --net ghactions_mm-test appropriate/curl:latest sh -c "until curl --max-time 5 --output - http://mysql:3306; do echo waiting for mysql; sleep 5; done;"
+          docker run --net ghactions_mm-test appropriate/curl:latest sh -c "until curl --max-time 5 --output - http://elasticsearch:9200; do echo waiting for elasticsearch; sleep 5; done;"
+      - name: Initialize the database with the source DB dump
+        run: |
+          curl ${{ inputs.db-dump-url }} | zcat | docker exec -i ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS
+      - name: Pull EE image
+        run: |
+          docker pull $BUILD_IMAGE
+      - name: Run migration through server
+        run: |
+          mkdir -p client/plugins
+          cd build
+          # Run the server in the background to trigger the migrations
+          docker run --name mmserver \
+            --net ghactions_mm-test \
+            --ulimit nofile=8096:8096 \
+            --env-file=dotenv/test.env \
+            --env MM_SQLSETTINGS_DRIVERNAME="mysql" \
+            --env MM_SQLSETTINGS_DATASOURCE="mmuser:mostest@tcp(mysql:3306)/mattermost_test?charset=utf8mb4,utf8&multiStatements=true" \
+            -v ~/work/mattermost-server:/mattermost-server \
+            -w /mattermost-server/mattermost-server \
+            $BUILD_IMAGE &
+          # In parallel, wait for the migrations to finish.
+          # To verify this, we check that the server has finished the startup job through the log line "Server is listening on"
+          until docker logs mmserver | grep "Server is listening on"; do\
+            echo "Waiting for migrations to finish..."; \
+            sleep 1; \
+            done;
+          # Make sure to stop the server. Also, redirect output to null;
+          # otherwise, the name of the container gets written to the console, which is weird
+          docker stop mmserver > /dev/null
+      - name: Cleanup DB
+        run : |
+          cd scripts/esrupgrades
+          docker exec -i ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS < $CLEANUP_SCRIPT
+      - name: Dump upgraded database
+        run: |
+          # Ignore tables containing meta-information about the migrations (db_{lock,migrations}),
+          # since those will not be present in the upgrade from the script.
```

Comment:
Should we remove this if it's now covered by the `$CLEANUP_SCRIPT`?

Reply:
Thanks! Done in https://github.com/mattermost/mattermost-server/pull/22573/commits/7552c5dd63d64b06e9cfd834edcf479490ead2de.